python/pipeline/codex/config.py

```diff
@@ -30,89 +36,161 @@ def _load_channel_names(data_dir):
 
 class CodexConfigV1(object):
 
-    def __init__(self, exp_config, processing_options, channel_names):
-        self.exp_config = exp_config
-        self.processing_options = processing_options
-        self.channel_names = channel_names
+    def __init__(self, conf):
+        self._conf = conf
 
-    def get_channel_names(self):
-        return self.channel_names
+    @property
+    def channel_names(self):
+        return self._conf['all_channel_names']
 
+    @property
     def n_cycles(self):
-        return self.exp_config['num_cycles']
+        return self._conf['num_cycles']
```

Comment:
Is there a reason why you have renamed all the variables (e.g. `num_cycles` -> `n_cycles`) instead of using the original names? Do they change their schema often?

Reply:
I waffled on that one a bit but given that I've already seen them change a couple times and that the naming conventions aren't consistent (per_pixel_XY_resolution, best_focus_channel, driftCompReferenceCycle, etc) I figured using the attributes directly would come back to bite me so instead I'm assuming there will be multiple versions of config classes that conform whatever they provide to the interface I expect.  

As for ```n_cycles``` instead of ```num_cycles``` there's not really a good reason for that other than that I was already renaming other things that made more sense to rename.

Reply:
Gotcha! Maybe we can adopt some type of json schema validator to lessen your headaches about these changes then. Perks of doing "cutting edge" bioinformatics ;)

Reply:
Ah like [this](https://github.com/eric-czech/codex/blob/r0.1/config/v0.1/schema/Experiment.json) (which is meant to work with the jsonschema validation lib)?  I started on it but got bored half-way through .. will try to finish that up eventually